vmicore/src/os/windows/KernelAccess.cpp

```diff
@@ -0,0 +1,229 @@
+uint32_t KernelAccess::extractExitStatus(addr_t eprocessBase) const
+{
+    return vmiInterface->read32VA(eprocessBase + kernelOffsets.eprocess.ExitStatus, vmiInterface->getSystemCr3());
+}
+
+addr_t KernelAccess::extractSectionAddress(addr_t eprocessBase) const
+{
+    return vmiInterface->read64VA(eprocessBase + kernelOffsets.eprocess.SectionObject, vmiInterface->getSystemCr3());
+}
+
+addr_t KernelAccess::extractControlAreaAddress(addr_t sectionAddress) const
+{
+    expectSaneKernelAddress(sectionAddress, __func__); // NOLINT(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+    return vmiInterface->read64VA(sectionAddress + kernelOffsets.section.controlArea, vmiInterface->getSystemCr3());
+}
+
+addr_t KernelAccess::extractControlAreaFilePointer(addr_t controlAreaAddress) const
+{
+    expectSaneKernelAddress(controlAreaAddress,
+                            __func__); // NOLINT(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+    return vmiInterface->read64VA(controlAreaAddress + kernelOffsets.controlArea.FilePointer,
+                                  vmiInterface->getSystemCr3());
+}
+
+std::unique_ptr<std::string> KernelAccess::extractProcessPath(addr_t filePointerAddress) const
+{
+    expectSaneKernelAddress(filePointerAddress,
+                            __func__); // NOLINT(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+    return vmiInterface->extractUnicodeStringAtVA(filePointerAddress + kernelOffsets.fileObject.FileName,
+                                                  vmiInterface->getSystemCr3());
+}
+
+uint32_t KernelAccess::extractFlagValue32(addr_t flagBaseVA, size_t startBit, size_t endBit) const
+{
+    expectSaneKernelAddress(flagBaseVA, __func__); // NOLINT(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+    auto flagValue = vmiInterface->read32VA(flagBaseVA, vmiInterface->getSystemCr3());
+
+    return getFlagValue(flagValue, startBit, endBit);
+}
+
+addr_t KernelAccess::getMmVadShortFlagsAddr(addr_t vadShortBaseVA) const
+{
+    return vadShortBaseVA + kernelOffsets.mmVadShort.Flags;
+}
+
+uint32_t KernelAccess::extractProtectionFlagValue(addr_t vadShortBaseVA) const
+{
+    return extractFlagValue32(getMmVadShortFlagsAddr(vadShortBaseVA),
+                              kernelOffsets.mmvadFlags.protection.startBit,
+                              kernelOffsets.mmvadFlags.protection.endBit);
+}
+
+bool KernelAccess::extractPrivateMemoryFlagValue(addr_t vadShortBaseVA) const
+{
+    return static_cast<bool>(extractFlagValue32(getMmVadShortFlagsAddr(vadShortBaseVA),
+                                                kernelOffsets.mmvadFlags.privateMemory.startBit,
+                                                kernelOffsets.mmvadFlags.privateMemory.endBit));
+}
+
+addr_t KernelAccess::getMmSectionFlagsAddr(addr_t controlAreaBaseVA) const
+{
+    return controlAreaBaseVA + kernelOffsets.controlArea._mmsection_flags;
+}
+
+bool KernelAccess::extractBeingDeletedFlagValue(addr_t controlAreaBaseVA) const
```

Comment:
nit: bool returning functions normally have question like names (for example: IsBeingDeleted).